wgengine/watchdog.go

```diff
@@ -66,6 +96,21 @@ func (e *watchdogEngine) watchdogErr(name string, fn func() error) error {
 		buf := new(strings.Builder)
 		pprof.Lookup("goroutine").WriteTo(buf, 1)
 		e.logf("wgengine watchdog stacks:\n%s", buf.String())
+
+		// Opportunistically try to print the currently in-flight
+		// operations, but don't block to avoid a deadlock.
+		if e.inFlightMu.TryLock() {
```

Comment:
why would you expect this mutex to be contended?

its critical section is tiny, no?

Reply:
I have no idea why I thought that. Moved to a regular `Lock` instead of `TryLock`.